frontend/src/context/useCurrentLocation.js

```diff
@@ -0,0 +1,100 @@
+import { useEffect, useReducer } from 'react'
+import axios from 'axios';
+import constate from 'constate'
+import logger from './Logger'
+import useLocalStorage from '../shared/useLocalStorage'
+
+
+const initialState = {
+  selectedLocation: {},
+  coordinates: []
+};
+
+
+const reducer = (originalState, action) => {
+  let state = Object.assign({}, originalState);
+  switch (action.type) {
+    case 'SET_SELECTED_COORDINATES':
+      return {
+        ...state,
+        coordinates: action.payload.coordinates,
+        selectedLocation: {},
+      }
+
+    case 'SET_SELECTED_LOCATION': 
+      const { suggestion, coordinates } = action.payload;
+      return {
+        ...state,
+        selectedLocation: suggestion,
+        coordinates,
+      };
+
+    default: {
+      return state;
+    }
+  }
+
+};
+
+const loggerReducer = logger(reducer);
+
+const useCurrentLocation = () => {
+  //  TODO: loader status: 
+  // const [isGettingLocation, setIsGettingLocation] = useState(false);
+  const [data, setData] = useLocalStorage('locationData', initialState)
```

Comment:
I think there is no need for this to be stored in LocalStorage, you can just use `useState`

Reply:
Well, it depends. we can use `useState` and later if it occurred we need persistence. we can refactor the part.